docker/base-deps/Dockerfile

```diff
@@ -0,0 +1,16 @@
+# The deploy Docker image build a self-contained Ray instance suitable
+# for end users.
+
+FROM ubuntu:xenial
+RUN apt-get update
+RUN apt-get install -y vim git wget
+RUN apt-get install -y cmake build-essential autoconf curl libtool libboost-all-dev unzip
```

Comment:
I believe best practice for Dockerfiles is to combine stuff into one big command and remove the temporary files, e.g., something like the following (taken from https://github.com/openai/gym/blob/master/Dockerfile)

```
RUN apt-get update \
    && apt-get install -y libav-tools \
    python-numpy \
    python-scipy \
    python-pyglet \
    python-setuptools \
    libpq-dev \
    libjpeg-dev \
    curl \
    cmake \
    swig \
    python-opengl \
    libboost-all-dev \
    libsdl2-dev \
    wget \
    unzip \
    git \
    xpra \
    && apt-get clean \
    && rm -rf /var/lib/apt/lists/* \
    && easy_install pip
```